lib/Sema/CSGen.cpp

```diff
@@ -3667,35 +3667,54 @@ void ConstraintSystem::optimizeConstraints(Expr *e) {
   e->walk(optimizer);
 }
 
-bool swift::isExtensionApplied(DeclContext &DC, Type BaseTy,
-                               const ExtensionDecl *ED) {
-  if (!ED->isConstrainedExtension() ||
-      // We'll crash if we leak type variables from one constraint
+bool swift::isGenericContextApplied(DeclContext *DC, Type BaseTy,
```

Comment:
I would suggest renaming this to areGenericRequirementsSatisfied() and having it take a GenericSignature